src/ol/layer/MapboxVector.js

```diff
@@ -155,8 +159,8 @@ const SourceType = {
  * style created with Mapbox Studio and hosted on Mapbox, this will look like
  * 'mapbox://styles/you/your-style'.
  * @property {string} [accessToken] The access token for your Mapbox style. This has to be provided
- * for `mapbox://` style urls. For `https://` and other urls, access keys must be part of the style
- * url.
+ * for `mapbox://` style urls. For `https://` and other urls, any access key must be the last query
+ * parameter of the style url.
```

Comment:
This answered my question above (`accessToken` only works for `mapbox://` style URLs).